unconvert_test.go

```diff
@@ -0,0 +1,171 @@
+// Copyright 2019 The Go Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package main_test
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"sort"
+	"strconv"
+	"strings"
+	"testing"
+)
+
+func TestBinary(t *testing.T) {
+	exepath, cleanup := build(t)
+	defer cleanup()
+
+	output, err := exec.Command(exepath, "./testdata").CombinedOutput()
+	if err == nil {
+		t.Fatal("expected to quit with an error code")
+	}
+
+	got, err := ParseOutput("testdata", string(output))
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	expected, err := ParseDir("testdata")
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	SortAnnotations(got)
+	SortAnnotations(expected)
```

Comment:
It looks like this is doing a deep equals comparison of got vs expected as lists of error messages. That will behave correctly, but I worry it won't make tracking down failures very easy. For example, if there's a single missing `//@` annotation, ideally there's only one error message. But I suspect if you remove the first one from input_test.go, you'll get a lot more than that.

I'd suggest doing a set difference approach like $GOROOT/test/run.go does: warn about any unexpected values in got that aren't in expected, and any missing values in expected that aren't in got. I.e., don't worry about ordering or length.

Reply:
Should be better now.